src/amoc_coordinator_worker.erl

```diff
@@ -0,0 +1,127 @@
+-module(amoc_coordinator_worker).
+
+-behaviour(gen_server).
+
+%% API
+-export([start/1, add/2,
+         stop/1, reset/1,
+         timeout/1]).
+
+%% gen_server callbacks
+-export([init/1,
+         handle_call/3,
+         handle_cast/2]).
+
+-type event() :: amoc_coordinator:coordination_event().
+-type action() :: amoc_coordinator:coordination_action().
+-type data() :: amoc_coordinator:coordination_data().
+
+-record(state, {required_n = all :: pos_integer() | all,
+                n = 0 :: non_neg_integer(),
+                actions = [] :: [action()],
+                collect_data = true :: boolean(),
+                accumulator = [] :: [data()]}).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec(start(amoc_coordinator:normalized_coordination_item()) -> {ok, Pid :: pid()}).
+start(CoordinationItem) -> gen_server:start(?MODULE, CoordinationItem, []).
```

Comment:
Could you explain why we use `start` and not `start_link` here?

Reply:
it should be started at the Scenario:init(), AFAIR that process is shortlived. And we should use it later from Scenario:start() context.

Reply:
And do we care if we loose the `amoc_coordinator_worker` process while the test is running?

Reply:
sorry, i got i wrong. yes, this one can and should be changed to `start_link`. it's important to have `gen_event` to be just started, but not linked.